server/text_generation_server/layers/moe/__init__.py

```diff
@@ -218,12 +219,17 @@ def __init__(
         down_proj_name: str = "down_proj",
     ):
         super().__init__()
-
         if (
             isinstance(weights.loader, DefaultWeightsLoader)
             and isinstance(weights.loader.weight_class, UnquantizedWeight)
         ) or isinstance(weights.loader, HybridFP8UnquantLoader):
-            cls = UnquantizedSparseMoELayer
+            if (
+                isinstance(weights.loader, HybridFP8UnquantLoader)
+                and weights.loader.to_fp8
+            ):
+                cls = FP8SparseMoELayer
+            else:
+                cls = UnquantizedSparseMoELayer
```

Comment:
I think it would be better to flatten this now. Something like:

```
if isinstance(weights.loader, DefaultWeightsLoader) and isinstance(weights.loader.weight_class, UnquantizedWeight):
    cls = UnquantizedSparseMoELayer
elif isinstance(weights.loader, HybridFP8UnquantLoader):
    cls = FP8SparseMoELayer
elif #...
```

Reply:
I flattened it, but the condition remains the same, because we always use the `HybridFP8UnquantLoader` to load the weigths. 
https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/utils/quantization.py#L202